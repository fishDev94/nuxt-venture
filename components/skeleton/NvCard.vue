<template>
  <div class="nv-skeleton-card-wrapper">
    <UCard class="nv-skeleton-card">
      <template #header>
        <USkeleton class="h-34 lg:h-48 rounded-b-none" />
      </template>

      <template #default>
        <div class="nv-skeleton-card__footer-container">
          <div class="nv-skeleton-card__footer-subhead">
            <USkeleton class="h-6 rounded-none" />
            <div class="flex flex-col gap-[4px]">
              <USkeleton class="w-[272px] h-3 rounded-none" />
              <USkeleton class="w-[240px] h-3 rounded-none" />
              <USkeleton class="w-[180px] h-3 rounded-none" />
            </div>
          </div>
          <div class="nv-skeleton-card__info">
            <USkeleton class="w-[60px] h-4 rounded-none" />
            <USkeleton class="w-[60px] h-4 rounded-none" />
          </div>
        </div>
      </template>
    </UCard>
  </div>
</template>

<style lang="scss" scoped>
@use "assets/style/utils" as *;

.nv-skeleton-card-wrapper {
  display: flex;
  flex: 0 0 230px;
  scroll-snap-align: start;

  @include start-from(tablet) {
    min-width: 320px;
  }
}

.nv-skeleton-card {
  display: flex;
  flex-direction: column;
  overflow: hidden;

  & > * {
    flex: 1;
  }

  &__footer-subhead {
    display: flex;
    flex-direction: column;
    gap: 6px;

    @include start-from(medium-desktop) {
      gap: 12px;
    }
  }

  &__footer-container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    gap: 8px;
    height: 100%;

    @include start-from(medium-desktop) {
      gap: 12px;
    }
  }

  &__info {
    display: flex;
    justify-content: space-between;
    align-items: center;

    &--price {
      @include body(5);

      @include start-from(medium-desktop) {
        @include body(4);
      }
    }
  }
}
</style>
