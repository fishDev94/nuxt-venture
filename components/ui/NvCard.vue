<template>
  <UCard class="nv-card" as="a" :href="`/id/${id}`">
    <template #header>
      <NuxtImg :src class="nv-card__img h-34 lg:h-48" />
    </template>

    <template #default>
      <div class="nv-card__footer-container">
        <div class="nv-card__footer-subhead">
          <h3 class="nv-card__title">{{ title }}</h3>
          <p class="nv-card__description">
            {{ description }}
          </p>
        </div>
        <div class="nv-card__info">
          <p class="nv-card__info--price">
            <strong>â‚¬ {{ price }}</strong>
          </p>
          <UiRatingStar v-if="rating" :rating />
        </div>
      </div>
    </template>
  </UCard>
</template>

<script lang="ts" setup>
import { PRICE_PLACEHOLDER } from "~/constants";

const {
  price = PRICE_PLACEHOLDER,
  title = "Title card placeholder",
  src = "https://picsum.photos/200/300",
  rating = undefined,
  description = "Lorem, ipsum dolor sit amet consectetur adipisicing elit. Laudantium quod, iure debitis maiores laborum dicta.",
} = defineProps<{
  title?: string;
  description?: string;
  src?: string;
  rating?: number;
  id: number;
  price?: number;
}>();
</script>

<style lang="scss" scoped>
@use "assets/style/utils" as *;

.nv-card {
  display: flex;
  flex-direction: column;
  overflow: hidden;
  flex: 0 0 230px;
  scroll-snap-align: start;

  @include start-from(tablet) {
    min-width: 320px;
  }

  & > * {
    flex: 1;
  }

  &__img {
    width: 100%;
  }

  &__footer-subhead {
    display: flex;
    flex-direction: column;
    gap: 6px;

    @include start-from(medium-desktop) {
      gap: 12px;
    }
  }

  &__footer-container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    gap: 8px;
    height: 100%;

    @include start-from(medium-desktop) {
      gap: 12px;
    }
  }

  &__title {
    @include body(3);

    font-weight: 700;

    @include start-from(medium-desktop) {
      @include body(2);

      font-weight: 700;
    }
  }

  &__description {
    @include body(6);

    @include start-from(medium-desktop) {
      @include body(5);
    }
  }

  &__info {
    display: flex;
    justify-content: space-between;
    align-items: center;

    &--price {
      @include body(5);

      @include start-from(medium-desktop) {
        @include body(4);
      }
    }
  }
}
</style>
