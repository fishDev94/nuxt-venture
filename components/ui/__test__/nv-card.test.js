import { mount } from '@vue/test-utils';
import NvCard from '../NvCard.vue';
import { describe, expect, test } from 'vitest';

describe('NvCard component', () => {
    test('renders correctly', () => {
        const wrapper = mount(NvCard, {
            props: {
                price: 100,
                title: 'test card',
                src: 'https://test-vitest.vitest',
                rating: 4.5,
                description: 'test description',
                id: 5
            }
        })

        expect(wrapper.html()).matchSnapshot()
    })
})