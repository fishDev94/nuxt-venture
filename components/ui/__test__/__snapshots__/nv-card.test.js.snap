// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NvCard component > renders correctly 1`] = `"<routerlink data-v-5ddb1678=\\"\\" to=\\"/products/5\\" class=\\"nv-card-wrapper\\"></routerlink>"`;
